### PR TITLE
feat(greptimedb sink): improve tls support for greptimedb sink

### PR DESCRIPTION
**Pull request by sunng87**
This patch improves TLS support for greptimedb sink, by updating the client sdk, and improving how we parse the configuration.

Before this patch, we will need to configure a custom server ca certificate to use TLS. However, for those hosting greptimedb using public ca, like letsencrypt, this server ca file is not required. So with change, we make server certificate completely optional for enable TLS.

This has been verified against greptimedb's hosted service.

### CODE REVIEW

**Review by jszwedko — APPROVED** [review]
Thanks @sunng87 !

**Comment by sunng87** [comment]
@jszwedko Sure! I have included a fragment for this feature.

File: changelog.d/20006_improve_greptimedb_tls.enhancement.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Improves TLS support for greptimedb sink. `tls.ca_file` is no longer required for enabling TLS. Just use `tls = {}` in toml configuration when your server is hosting a public CA.
```
Comment (jszwedko) [inline changelog.d/20006_improve_greptimedb_tls.enhancement.md:1]:
```suggestion
Improves TLS support for greptimedb sink. `tls.ca_file` is no longer required for enabling TLS. Just use `tls = {}` in toml configuration when your server is hosting a public CA.

authors: sunng87
```

**Review by jszwedko — APPROVED** [review]
I think for external contributors it requires the authors field. Thanks for adding the fragment!

**Review by jszwedko — APPROVED** [review]

**Pull request merged**